hs_core/tests/api/utils.py

```diff
@@ -19,21 +18,14 @@ def __init__(self, file=None, name=None, content_type=None, size=None, charset=N
     def temporary_file_path(self):
         return self.orig_name
 
-def zip_up(ziph, root_directory, directory=""):
-    full_path = Path(os.path.join(root_directory, directory))
-    dirs = [str(item) for item in full_path.iterdir() if item.is_dir()]
-    files = [str(item) for item in full_path.iterdir() if item.is_file()]
-    for file in files:
-        ziph.write(file, arcname=os.path.join(directory, os.path.basename(file)))
-    for d in dirs:
-        zip_up(ziph, root_directory, os.path.join(directory, os.path.basename(d)))
-
 
 def prepare_resource(folder, res, user, extracted_directory, test_bag_path):
     from hs_core.views.utils import unzip_file
 
-    zipf = zipfile.ZipFile(test_bag_path, 'w')
-    zip_up(zipf, os.path.join(extracted_directory, folder))
+    dir_to_zip = os.path.join(extracted_directory, folder)
+    # remove '.zip' extension from the zip file path as the extension will be added by make_archive call
+    zip_to_file_path = test_bag_path[:-4]
+    shutil.make_archive(zip_to_file_path, 'zip', dir_to_zip)
```

Comment:
I am able to create a folder containing a prohibited char and then zip it up within the FB. Should this be allowed?

Reply:
One should not be able to create a folder with a prohibited character. I have tests covering each of the prohibited characters to test that folder creation fails. If you were able to create a folder with prohibited character then there is a bug in my code. Do you remember which character you used?

Reply:
Sorry @pkdash my comment was misleading. I said I was "able to create a folder" but actually this was with a folder that existed prior to checking out your PR. If the folder already exists, I'm able to zip it up, but I can't unzip it:
https://user-images.githubusercontent.com/17934193/203173864-55e98502-ed10-4ae8-902a-728db7281fa7.mov
but this isn't much of an issue...